hatch_build.py

```diff
@@ -0,0 +1,83 @@
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+from hatchling.metadata.plugin.interface import MetadataHookInterface
+
+import tomlkit
+import pathlib
+
+
+class PinVersionMetadataHook(MetadataHookInterface):
+    """Update the dependency versions so they are pinned metadata of the wheel artefact produced by a wheel."""
+    def update(self, metadata):
+        change_count = 0
+        # Overlay the pinned dependencies, replacing dependencies if they already exist.
+        for pinned_dep in self.config["config"]["pinned_dependencies"]:
+            pinned_dep_name = pinned_dep.split(" ")[0]
+            for dep in metadata["dependencies"]:
+                dep_name = dep.split(" ")[0]
+                # If the dependency is already in the list, replace it with the pinned version.
+                if dep_name == pinned_dep_name:
```

Comment:
In the 'METADATA' file, it looks like the new pinned version chunk is appearing twice.

Reply:
Fixed in most recent commit.